src/ports/server/component.ts

```diff
@@ -251,7 +251,27 @@ export async function createServerComponent({
     app.use(bodyParser.json())
 
     // CORS middleware
-    app.use(cors({ origin: corsOrigin, methods: corsMethods }))
+    const corsOptions = {
```

Comment:
The CORS middleware can accept various types of origins, one of them is an array of regexes, which eventually ends up performing something similar as the origin function you've crafted here.
What do you think about using something like what we've done [here](https://github.com/decentraland/marketplace-server/blob/16cd8b1b612745d7db789c8b5bf10c2cc178574a/src/components.ts#L31)? It's simpler, although it would only accept regexes, but we can change the default `.env` to have a regex as well, like [this](https://github.com/decentraland/marketplace-server/blob/16cd8b1b612745d7db789c8b5bf10c2cc178574a/.env.default#L10) and it should work on the development side as well.